EpikCord/client/http_client.py

```diff
@@ -247,39 +249,34 @@ async def log_request(res, body: Optional[dict] = None):
         finally:
             logger.debug("".join(message))
 
-    async def get(  # type: ignore
+
+    def base(  # type: ignore
         self,
+        method,
         url,
         *args,
         to_discord: bool = True,
         **kwargs,
     ):
         if to_discord:
-            return await self.request("GET", url, *args, **kwargs)
-        return await super().get(url, *args, **kwargs)
-
-    async def post(self, url, *args, to_discord: bool = True, **kwargs):  # type: ignore
-        if to_discord:
-            return await self.request("POST", url, *args, **kwargs)
-        return await super().post(url, *args, **kwargs)
-
-    async def patch(self, url, *args, to_discord: bool = True, **kwargs):  # type: ignore
-        if to_discord:
-            res = await self.request("PATCH", url, *args, **kwargs)
-            return res
-        return await super().patch(url, *args, **kwargs)
-
-    async def delete(self, url, *args, to_discord: bool = True, **kwargs):  # type: ignore
-        if to_discord:
-            res = await self.request("DELETE", url, *args, **kwargs)
-            return res
-        return await super().delete(url, **kwargs)
-
-    async def put(self, url, *args, to_discord: bool = True, **kwargs):  # type: ignore
-        if to_discord:
-            res = await self.request("PUT", url, *args, **kwargs)
-            return res
-        return await super().put(url, *args, **kwargs)
-
+            return self.request(method, url, *args, **kwargs)
+        return self.session.request(method, url, *args, **kwargs)
+
+    get  = partialmethod(base, "GET") 
+    post = partialmethod(base, "POST")
+    put = partialmethod(base, "PUT")
+    patch = partialmethod(base, "PATCH")
+    delete = partialmethod(base, "DELETE")
+    head = partialmethod(base, "HEAD")
+    options = partialmethod(base, "OPTIONS")
+
+
+    async def get_gateway(self) -> discord_typings.GetGatewayData:
+        res = await self.get("/gateway")
+        return await res.json()
+    
+    async def get_gateway_bot(self) -> discord_typings.GetGatewayBotData:
+        res = await self.get("/gateway/bot")
+        return await res.json()
```

Comment:
`/gateway` for non-sharded bots, `/gateway/bot` for sharded bots as it gives recommended shard count there

Reply:
They're different endpoints, one of them requires a token, which in turn provides more information about the bot.

Reply:
Hmm... is it necessary to have 2 method for this? Couldn't get a generic `get_gateaway` that will reduce the correct one automatically?

I mean like `"/gateway" + ('/bot' * isinstance(bot, ShardedBot)` or something like that

Reply:
that is also good

Reply:
i like to over engeneer x)

Reply:
The thing is that it's not specific for Sharded bots, it contains the amount of times the bot has tried to login, which can be useful for somethin like a `stats` command, and overall this provides flexibility to the Developer

Reply:
> The thing is that it's not specific for Sharded bots, it contains the amount of times the bot has tried to login, which can be useful for somethin like a `stats` command, and overall this provides flexibility to the Developer

I see then, i guess it would be nice to have that specified in the docstring to clear out possible confusion

Reply:
Will do.